packages/expo-router/src/fork/extractPathFromURL.ts

```diff
@@ -43,26 +42,36 @@ function extractExactPathFromURL(url: string): string {
 }
 
 /** Major hack to support the makeshift expo-development-client system. */
-function isExpoDevelopmentClient(url: URL<Record<string, string | undefined>>): boolean {
+function isExpoDevelopmentClient(url: URL): boolean {
   return !!url.hostname.match(/^expo-development-client$/);
```

Comment:
Unrelated to this PR but shouldn't this just be:

```suggestion
  return url.hostname === 'expo-development-client';
```
?